client/my-sites/plugins/plugins-discovery-page/index.jsx

```diff
@@ -0,0 +1,207 @@
+import SingleListView from '../plugins-browser/single-list-view';
+import usePlugins from '../use-plugins';
+
+/**
+ * Module variables
+ */
+const SHORT_LIST_LENGTH = 6;
+
+const UpgradeNudge = ( { selectedSite, sitePlan, isVip, jetpackNonAtomic, siteSlug } ) => {
+	const translate = useTranslate();
+	const eligibleForProPlan = useSelector( ( state ) =>
+		isEligibleForProPlan( state, selectedSite?.ID )
+	);
+
+	if ( ! selectedSite?.ID || ! sitePlan || isVip || jetpackNonAtomic ) {
+		return null;
+	}
+	const isLegacyPlan = isBlogger( sitePlan ) || isPersonal( sitePlan ) || isPremium( sitePlan );
+	const checkoutPlan = eligibleForProPlan && ! isLegacyPlan ? 'pro' : 'business';
+	const bannerURL = `/checkout/${ siteSlug }/${ checkoutPlan }`;
+	const plan = findFirstSimilarPlanKey( sitePlan.product_slug, {
+		type: TYPE_BUSINESS,
+	} );
+
+	const title =
+		eligibleForProPlan && ! isLegacyPlan
+			? translate( 'Upgrade to the Pro plan to install plugins.' )
+			: translate( 'Upgrade to the Business plan to install plugins.' );
+
+	return (
+		<UpsellNudge
+			event="calypso_plugins_browser_upgrade_nudge"
+			showIcon={ true }
+			href={ bannerURL }
+			feature={ FEATURE_INSTALL_PLUGINS }
+			plan={ plan }
+			title={ title }
+		/>
+	);
+};
+
+const UpgradeNudgePaid = ( props ) => {
+	const translate = useTranslate();
+
+	const requiredPlans = useSelector( ( state ) =>
+		getPlansForFeature( state, props.selectedSite?.ID, WPCOM_FEATURES_INSTALL_PURCHASED_PLUGINS )
+	);
+
+	if ( ! requiredPlans ) {
+		return null;
+	}
+
+	const requiredPlan = getPlan( requiredPlans[ 0 ] );
+
+	return (
+		<UpsellNudge
+			event="calypso_plugins_browser_upgrade_nudge"
+			showIcon={ true }
+			href={ `/checkout/${ props.siteSlug }/${ requiredPlan.getPathSlug() }` }
+			feature={ WPCOM_FEATURES_INSTALL_PURCHASED_PLUGINS }
+			plan={ requiredPlan.getStoreSlug() }
+			title={ translate( 'Upgrade to the %(planName)s plan to install premium plugins.', {
+				textOnly: true,
+				args: { planName: requiredPlan.getTitle() },
+			} ) }
+		/>
+	);
+};
+
+function filterPopularPlugins( popularPlugins = [], featuredPlugins = [] ) {
+	const displayedFeaturedSlugsMap = new Map(
+		featuredPlugins
+			.slice( 0, SHORT_LIST_LENGTH ) // only displayed plugins
+			.map( ( plugin ) => [ plugin.slug, plugin.slug ] )
+	);
+
+	return popularPlugins.filter(
+		( plugin ) =>
+			! displayedFeaturedSlugsMap.has( plugin.slug ) && isCompatiblePlugin( plugin.slug )
+	);
+}
+
+const PluginsDiscoveryPage = ( props ) => {
+	const {
+		plugins: pluginsByCategoryFeatured = [],
+		isFetching: isFetchingPluginsByCategoryFeatured,
+	} = usePlugins( {
+		category: 'featured',
+	} );
+
+	const { plugins: popularPlugins = [], isFetching: isFetchingPluginsByCategoryPopular } =
+		usePlugins( {
+			category: 'popular',
+		} );
+
+	const {
+		plugins: pluginsBySearchTerm = [],
+		isFetching: isFetchingPluginsBySearchTerm,
+		pagination: pluginsPagination,
+		fetchNextPage,
+	} = usePlugins( {
```

Comment:
NITPICK: to make the code more readable can we find a way to not call `usePlugins` multiple times? Maybe we can create a component for each section it could live in this file or folder something like:
- `PopularSection`
- `PaidSection`
- `FeaturedSection`

And they will be responsible of calling the `usePlugins` hook to fetch it's data

Reply:
This can be done in a follow-up or added as a sub task of this one

Reply:
For whoever gets to create those...

`SingleListView` is also branching based on category passed in. I think it'd be good to replace `SingleListView` with your suggested category sections which in turn can just render customized `PluginsBrowserList`'s directly.

In other words, lets replace `SingleListView` with specialized `PopularSection`, `PaidSection`, `FeaturedSection` components.

Reply:
> In other words, lets replace SingleListView with specialized PopularSection, PaidSection, FeaturedSection components.

+1, Let's try to do this on this task

Reply:
This PR? Doesn't have to be. Just wanted to +1 your suggestion :smile:

Reply:
I just created `PaidPluginsSection` and `PopularPluginsSection`.
The popular section needs the Featured plugins to filter the popular, so I keep the featured plugins on the `PluginsDiscoveryPage` component and pass the `pluginsByCategoryFeatured` prop to the `PopularPluginsSection`

Reply:
Good compromise!

Reply:
Nicely done @lupus2k thanks a lot for that!